apps/wallet/src/ui/app/staking/stake/useStakingMutation.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { type SuiAddress, type SuiMoveObject } from '@mysten/sui.js';
+import { useMutation } from '@tanstack/react-query';
+
+import { useSigner, useAppSelector } from '_hooks';
+import { ownedObjects } from '_redux/slices/account';
+import { Coin } from '_redux/slices/sui-objects/Coin';
+
+interface StakeTokenArgs {
+    tokenTypeArg: string;
+    amount: bigint;
+    validatorAddress: SuiAddress;
+}
+
+export function useStakeTokenMutation() {
```

Comment:
A couple things with this:
- I'd actually be okay if this was just defined inline and not in a separate file.
- This file is named `useStakingMutation` but that hook isn't actually exposed. Maybe rename to something like `stakingHooks`?